cfn.template

```diff
@@ -189,16 +201,96 @@ Resources:
                     # Set permissions of artefacts
                     objects=$(aws s3 ls --recursive "s3://${BUCKET}/${project}/${version}")
                     echo "$objects" | awk -v bucket="$BUCKET" -v acl="$acl" '{system("aws s3api put-object-acl --bucket " bucket " --key " $4 " --acl " acl)}'
+                  done
+
+                  # Create an index file in the first region's bucket
+                  for region in $regions; do
+                    first_region="$region"
+                    break
+                  done
+
+                  cat <<EOF >index.html
+                  <!DOCTYPE html>
+                  <html lang="en">
+                    <head>
+                      <title>${project}</title>
+                      <style>
+                        table {
+                          border-collapse: collapse;
+                        }
+                        table, th, td {
+                          border: 1px solid black;
+                        }
+                        th, td {
+                          text-align: center;
+                          padding: 0.5em;
+                        }
+                        p.footer {
+                          font-size: 75%;
+                        }
+                      </style>
+                    </head>
+                    <body>
+                      <h1>${project}</h1>
+                      <table>
+                        <tr>
+                          <th>Region</th>
+                  EOF
+
+                  for template in $templates; do
+                  cat <<EOF >>index.html
+                          <th>$(basename $template)</th>
+                  EOF
+                  done
+
+                  cat <<EOF >>index.html
+                        </tr>
+                  EOF
+
+                  for region in $regions; do
+                    export AWS_DEFAULT_REGION="${region}"
+                    export BUCKET="${bucket_name_prefix}-${region}"
+
+                    cat <<EOF >>index.html
+                        <tr>
+                          <th>${region}</th>
+                  EOF
 
-                    # Sync to latest
-                    if [ "$version" != "latest" ]; then
-                      echo "Syncing $version to latest"
-                      aws s3 rm --recursive "s3://${BUCKET}/${project}/{latest}"
-                      aws s3 cp --acl "$acl" "s3://${BUCKET}/${project}/${version}/main.template" "s3://${BUCKET}/${project}/latest/main.template"
-                    fi
+                    for template in $templates; do
+                      name="${template%.*}"
```

Comment:
I'm not sure about this. Previously, we were setting stack name to the project name. Now that we can have multiple templates, we clearly can't do that.

Now I'm using the template filename (minus extension) as the stack name but that could just result in stacknames like `cfn` and `template`.

I consider concatenating the project name and the template name but those could get long and upset cfn's stack name limit.

Reply:
there's probably a fairly decent chance of a template filename containing an underscore which isn't a supported CloudFormation stack name character (alphanum + hyphens). the character limit on stack names is 128 chars so I don't think it's particularly likely that it'll be too long?

we don't actually address the use of invalid characters today, but I think the probability of project name including underscores is less than that of the filename including them?

Reply:
Well, we could convert unsupport characters into hyphens, that's easy enough. I'll add that but it still doesn't address the problem of whether to use `{project}-{filename}` or `{project}-{template number}` or just `{filename}` :S